OA-PASS/pass-docker
# Update Notification Services with BCC support

# About
This is a SNAPSHOT release of Notification Services that includes BCC support.  It updates the configuration file for notification services so that BCC addresses may be supported via ENV vars:

`PASS_NOTIFICATION_DEMO_GLOBAL_BCC_ADDRESS`
`PASS_NOTIFICATION_PRODUCTION_GLOBAL_BCC_ADDRESS`

The image included in this PR may be deployed to test, but should not be deployed to demo.  After deployment to test, an official release of Notification Services will follow, along with a production image.

LDAP image is updated with additional users supporting the testing of blind carbon copy of notifications in Notification Service integration tests.
- Add additional sample users to support logins to the mail server for blind carbon copy accounts.  - Required by the lasted version of Notification Services

# How to test

1.  Start pass-docker.
1.  Tail the logs of the `notification` container (`docker logs -f notification`)
1.  Prepare a submission on behalf of someone else
1.  See the notification appear in the logs, and note the BCC address

If you want to actually see the content of the notification in the BCC addressee's inbox:
1.  Obtain a shell in the `mail` container (`docker exec -ti mail /bin/bash`)
1.  Navigate to the **BCC** user email directory`/var/spool/mail/jhu.edu/notification-demo-bcc/new`

Every file in the directory will contain an email message.  The file name is based on the SMTP message ID.  If you only prepared a single submission, then there'll be one file containing the BCC notification.  If you prepared two submissions, there'll be two files, and so on.

Look at the content of the file, note the headers: you are looking at the **BCC** email, but you'll see that the BCC user is not shown in the headers.

Other notable aspects are the presence of X-* headers containing the notification type and the submission resource uri.
